### PR TITLE
fix(schematics): avoid multiple async calls

### PR DESCRIPTION
**Pull request by amitport**

### CODE REVIEW

File: src/lib/schematics/nav/files/__path__/__name@dasherize@if-flat__/__name@dasherize__.component.ts [hunk 1]
```diff
@@ -16,13 +16,25 @@ import { map } from 'rxjs/operators';
...
 
-  constructor(private breakpointObserver: BreakpointObserver) {}
+  /** Emits when the component is destroyed. */
+  private readonly _destroyed = new Subject<void>();
+
+  constructor(breakpointObserver: BreakpointObserver) {
+    breakpointObserver.observe(Breakpoints.Handset)
+      .pipe(
+        takeUntil(this._destroyed)
+      ).subscribe(result => {
```
Comment (DevVersion) [inline src/lib/schematics/nav/files/__path__/__name@dasherize@if-flat__/__name@dasherize__.component.ts]: Can we do something like:

```ts
breakpointObserver.observe(Breakpoints.Handset)
  .pipe(takeUntil(this._destroyed))
  .subscribe(result => this.isHandset = result.matches);
```

Basically the same code, just a bit more compact and more readable IMO.

Comment (amitport) [reply src/lib/schematics/nav/files/__path__/__name@dasherize@if-flat__/__name@dasherize__.component.ts]: I don't mind removing spaces, but `{}` make sense to me.. It signifies that the arrow function does not return a value and has a side-effect.

Would this be OK?:
```
breakpointObserver.observe(Breakpoints.Handset)
  .pipe(takeUntil(this._destroyed))
  .subscribe(result => {this.isHandset = result.matches;});
```

**Comment by jelbourn** [comment]
I'm not sure this is the right thing for the schematic. Generally the Angular team considers using the `async` pipe the best practice. Since the schematics are a starting point for many developers, we want to make sure they're following that. 

cc @robwormald for input

**Comment by amitport** [comment]
async pipe is best practice? in most cases it really isn't. When telling developers to use async pipe you need them to understand the exceptions (cold observers etc,.). It's much more practical to avoid using aync pipe when there is more than 1 subscription (no exceptions). People without much experience with Angular can manage that without breaking things.

**Comment by jelbourn** [comment]
@amitport this PR was accidentally merged and thus reverted in #13399 ; feel free to reopen a new PR with the change

**Pull request merged**